runtime/compiler/ilgen/Walker.cpp

```diff
@@ -2236,17 +2236,27 @@ TR_J9ByteCodeIlGenerator::genCheckCast()
 void
 TR_J9ByteCodeIlGenerator::genCheckCast(int32_t cpIndex)
    {
-   if (TR::Compiler->om.areValueTypesEnabled()
-       && TR::Compiler->cls.isClassRefPrimitiveValueType(comp(), method()->classOfMethod(), cpIndex))
+   bool isClassRefPrimitiveValueType = false;
+   if (TR::Compiler->om.areValueTypesEnabled())
       {
-      TR::Node * objNode = _stack->top();
+      if (TR::Compiler->om.doNotSupportQDescriptorForValueTypes())
+         {
+         TR_ASSERT_FATAL(false, "TO BE IMPLEMENTED!!!");
+         }
```

Comment:
I feel it'd be better to keep it for now in case of the spec changes.  We will reevaluate it when we implement the case where Q signature is not supported (`!TR::Compiler->om.isQDescriptorForValueTypesSupported()`)

Reply:
Addressed in c4346cc